metrics/internal/cache/pv.go

```diff
@@ -72,9 +117,67 @@ func (p *PersistentVolumeStore) Add(obj interface{}) error {
 		Pool:                           pv.Spec.CSI.VolumeAttributes["pool"],
 	}
 
+	clients, err := runCephRBDStatus(&p.monitorConfig, pv.Spec.CSI.VolumeAttributes["pool"], pv.Spec.CSI.VolumeAttributes["imageName"])
+	if err != nil {
+		return fmt.Errorf("failed to get image status %v", err)
+	}
+
+	klog.Info("getting name for the node here: ")
+
+	nodeName, err := getNodeNameForPV(pv, p.kubeClient)
+	if err != nil {
+		klog.Errorf("failed to get node IP for pod: %v", err)
+		return nil
+	}
+
+	klog.Info(clients)
+	if len(clients.Watchers) != 0 {
+		p.RBDClientMap[nodeName] = clients
```

Comment:
need a mutex lock here.

also, since nodes might be replaced by new nodes with same name, should we map to node UID here?